### PR TITLE
Garrett

### LINKED ISSUE #4: Complete Step 3: JSON

### LINKED ISSUE #8: Add Translate Library

### PR DESCRIPTION
**Pull request by glanden**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -23,10 +30,20 @@
...
+  private ArrayList<String> comments = new ArrayList<String>();
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Garrett!</h1>");
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+    comments.add(text);
+    Entity commentEntity = new Entity("Comment");
```
Comment (chyheimjb) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:39]: FYI. Using "Comment" as your key will give each comment you create the same key. You can use KeyFactory to make unique keys for each comment. 

https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/KeyFactory#createKey-java.lang.String,%20long-

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+
```
Comment (chyheimjb) [inline portfolio/src/main/webapp/script.js:1]: remove this whitespace

**Review by chyheimjb — APPROVED** [review]

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -11,6 +13,37 @@
...
 // See the License for the specific language governing permissions and
 // limitations under the License.
+/**
+ * Fetches message from the server and adds it to the DOM with translation.
+ */
+function getData() {
+  console.log('Fetching data.');
+  const languageCode = document.getElementById('language').value;
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/list-comments?language='+languageCode).then(response => response.json())
```
Comment (Sunlis) [inline portfolio/src/main/webapp/script.js:23]: You don't need to save this promise as a variable, just calling the method is fine.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -11,6 +13,37 @@
...
+  .then((comments) => { 
+  const commentsListElement = document.getElementById('server-comments-container');
```
Comment (Sunlis) [inline portfolio/src/main/webapp/script.js:25]: The indentation got a little messy here. It helps your brain navigate code if you keep things organized.
```suggestion
  fetch('/list-comments?language=' + languageCode)
      .then(response => response.json())
      .then((comments) => {
        const commentsListElement = document.getElementById('server-comments-container');
        // ...
      });
```

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -11,6 +13,37 @@
...
+  commentsListElement.innerHTML = '';
+  var i;
+  for(i = 0; i<comments.length; i++){
```
Comment (Sunlis) [inline portfolio/src/main/webapp/script.js]: You can define this variable inside the for loop
```suggestion
  for(let i = 0; i<comments.length; i++){
```

File: portfolio/src/main/java/com/google/sps/servlets/ListServlet.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    PreparedQuery results = datastore.prepare(query);
+    String languageCode = request.getParameter("language");
+    List<String> comments = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String title = (String) entity.getProperty("text");
+      Translate translate = TranslateOptions.getDefaultInstance().getService();
+    Translation translation =
+        translate.translate(title, Translate.TranslateOption.targetLanguage(languageCode));
+    String comment = translation.getTranslatedText();
+      comments.add(comment);
```
Comment (Sunlis) [inline portfolio/src/main/java/com/google/sps/servlets/ListServlet.java:37]: Make sure you keep indentation consistent.
```suggestion
      Translation translation =
        translate.translate(title, Translate.TranslateOption.targetLanguage(languageCode));
      String comment = translation.getTranslatedText();
        comments.add(comment);
```

**Pull request merged**